workspaces/adventure-pack/goodies/python3/src/traverse_level_order/__init__.py

```diff
@@ -0,0 +1,30 @@
+from collections import deque
+from typing import Generator, Optional
+
+
+class TreeNode:
+    def __init__(
+        self,
+        val: int = 0,
+        left: Optional["TreeNode"] = None,
+        right: Optional["TreeNode"] = None,
+    ):
+        self.val = val
+        self.left = left
+        self.right = right
+
+
+def traverse_level_order(
+    root: Optional[TreeNode],
+) -> Generator[TreeNode, None, None]:
```

Comment:
For a level traversal, I'm thinking we should yield an entire level at a time, since if you picked this kind of traversal you want to operate on levels rather than individual nodes.

```suggestion
) -> Generator[List[TreeNode], None, None]:
```

Reply:
Interesting idea. Will update the goody!

Reply:
Changed to return a list of nodes instead of separate nodes, but decided to pre-filter `None`s for this goody. It looks cleaner to me.

Reply:
Agreed, filtering is a good idea here.